File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        pottable = 0
        prev = 0
        
        for i, pot in enumerate(flowerbed):
            match prev:
                case 0:
                    if pot == 0:
                        pottable += 1
                        prev = 2
                        continue
                case 2:
                    if pot == 1:
                        pottable -= 1
            prev = pot
        
        return pottable >= n
        

File: 0605-can-place-flowers/0605-can-place-flowers.rs
impl Solution {
    pub fn can_place_flowers(flowerbed: Vec<i32>, n: i32) -> bool {
        let mut pottable = 0;
        let mut prev = 0;
        
        for (i, pot) in flowerbed.iter().enumerate() {
            match prev {
                0 => {
                    if *pot == 0 {
                        prev = 2;
                        pottable += 1;
                        continue;
                    }
                }
                2 => {
                    if *pot == 1 {
                        pottable -= 1;
                    }
                }
                _ => {}
            }
            
            prev = *pot;
        }
        
        pottable >= n
    }
}